src/ai/backend/manager/models/alembic/versions/1d42c726d8a3_add_container_registry_tables.py

```diff
@@ -0,0 +1,304 @@
+                input_config["project"] = project
+                input_configs.append(input_config)
+        else:
+            input_configs.append(input_config_template)
+
+    if not input_configs:
+        print("There is no container registries data to migrate in etcd.")
+        return
+
+    ContainerRegistryRow = get_container_registry_row_schema()
+
+    db_connection = op.get_bind()
+    db_connection.execute(sa.insert(ContainerRegistryRow).values(input_configs))
+
+
+def revert_data_psql_to_etcd() -> None:
+    ContainerRegistryRow = get_container_registry_row_schema()
+
+    db_connection = op.get_bind()
+    rows = db_connection.execute(sa.select(ContainerRegistryRow)).fetchall()
+    items = []
+
+    for id, url, registry_name, type, project, username, password, ssl_verify, _is_global in rows:
+        item = {
+            "": url,
+            "type": str(type),
+            "hostname": registry_name,  # registry_name should be reverted to hostname,
+        }
+
+        if project is not None:
+            item["project"] = project
+        if username is not None:
+            item["username"] = username
+        if password is not None:
+            item["password"] = password
+        if ssl_verify is not None:
+            item["ssl_verify"] = ssl_verify
+
+        items.append(item)
+
+    # Records with the same hostname should be grouped together.
+    # Note that if there are columns other than the project column that have different values between records,
+    # information loss can occur.
+    grouped_items = {k: list(v) for k, v in groupby(items, key=lambda x: x["hostname"])}
+
+    def merge_items(items):
+        for item in items:
+            if "project" not in item:
+                return items[0]
+
+        projects = [item["project"] for item in items]
+        merged_projects = ",".join(projects)
+
+        merged_item = items[0].copy()
+        merged_item["project"] = merged_projects
+
+        return merged_item
+
+    merged_items = [merge_items(items) for items in grouped_items.values()]
+
+    def put_etcd_container_registries(merged_items: list[Any], queue: Queue):
+        etcd = get_async_etcd()
+        for item in merged_items:
+            hostname = item.pop("hostname")
+            asyncio.run(etcd.put_prefix(f"{ETCD_CONTAINER_REGISTRY_KEY}/{hostname}", item))
+
+        queue.put(True)
+
+    queue: Queue = Queue()
+
+    with ThreadPoolExecutor() as executor:
+        executor.submit(put_etcd_container_registries, merged_items, queue)
+
+    queue.get()
+
+
+def insert_registry_id_to_images() -> None:
+    db_connection = op.get_bind()
+    ContainerRegistry = get_container_registry_row_schema()
+
+    image_infos = db_connection.execute(sa.select([ImageRow.id, ImageRow.name])).fetchall()
+
+    for image_info in image_infos:
+        image_id, image_name = image_info
+
+        parts = image_name.split("/")
+        registry_name = parts[0]
+        project = parts[1]
```

Comment:
```suggestion
        registry_name, _, project = image_name.partition("/")
```
Not a big deal, but we prefer to use `partition` if we are sure that the string should be divided into two strings.

Reply:
We can't use `partition` like this.

For example, 
> image_name = "registry_name/project/something"
> image_name.partition("/")
('registry_name', '/', 'project/something')

'project/something' should be just project